pkgs/development/web/nodejs/nodejs.nix

```diff
@@ -229,20 +343,9 @@ let
 
       # assemble a static v8 library and put it in the 'libv8' output
       mkdir -p $libv8/lib
-      pushd out/Release/obj.target
+      pushd out/Release/obj
       find . -path "./torque_*/**/*.o" -or -path "./v8*/**/*.o" | sort -u >files
-      ${if stdenv.buildPlatform.isGnu then ''
-        ar -cqs $libv8/lib/libv8.a @files
-      '' else ''
-        # llvm-ar supports response files, so take advantage of it if it’s available.
-        if [ "$(basename $(readlink -f $(command -v ar)))" = "llvm-ar" ]; then
-          ar -cqs $libv8/lib/libv8.a @files
-        else
-          cat files | while read -r file; do
-            ar -cqS $libv8/lib/libv8.a $file
-          done
-        fi
-      ''}
+      $AR -cqs $libv8/lib/libv8.a @files
```

Comment:
Note: before this change, we had fallback for `ar` that does not support response files. As far as I know, that is only the case for Darwin’s cctools, but stdenv already uses LLVM for most of the tools on Darwin.
https://github.com/NixOS/nixpkgs/blob/034f3eec9b56e2fa22ee77f67403310c202ed398/pkgs/os-specific/darwin/binutils/default.nix#L117

That is, I think it makes sense to assume that `ar` can handle response files (unless there is some edge case I’m missing).

Reply:
In the future, ideally, we should update v8 package and avoid using nodejs as v8 library provider.

Reply:
You could set `env.AR` to `"${lib.getBin llvm}/bin/llvm-ar"` to play it really safe, but that’s not necessary.

Darwin bintools has been using `llvm-ar` since 23.11. Going forward, more tools will be switched to their LLVM versions as compatibility and stability improve. None are expected to switch back to their cctools versions.